arrow-libs/core/arrow-core/src/main/kotlin/arrow/core/Iterable.kt

```diff
@@ -329,6 +329,12 @@ fun <E, A> Iterable<Validated<E, A>>.sequenceValidated(semigroup: Semigroup<E>):
 fun <E, A> Iterable<ValidatedNel<E, A>>.sequenceValidated(): ValidatedNel<E, List<A>> =
   traverseValidated(Semigroup.nonEmptyList(), ::identity)
 
+inline fun <A, B> Iterable<A>.traverseOption(f: (A) -> Option<B>): Option<List<B>> =
+  traverseEither { f(it).toEither { Unit } }.orNone()
```

Comment:
```suggestion
inline fun <A, B> Iterable<A>.traverseOption(f: (A) -> Option<B>): Option<List<B>> {
  val acc = mutableListOf<B>()
  forEach { a ->
    when (val res = f(a)) {
      is Some -> acc.add(res.value)
      None -> return@traverseOption res
    }
  }
  return acc.some()
}
```

Nit: a direct implementation reads better IMO, and is (slightly) more performant.
That'll save 1 allocation per element in the `Iterable`.